okio/jvm/src/main/java/okio/OkioUpgrade.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+import java.net.Socket
+import java.nio.file.OpenOption
+import java.nio.file.Path
+
+val Okio = OkioUpgrade
```

Comment:
This is my sneakiest hack . . . callsites say `Okio.source()` but they’re dereferencing a package-level val called Okio, not a class. I’d prefer the class, but anything like that collides with the JVM name of `Okio.kt`, and there’s no `@JvmName` overrides for class names.

Reply:
We need to prevent Java callers from seeing the val and the object. More `@JvmName`? `@JvmSynthetic`?

Reply:
Might have to do it on the file to hide `OkioUpgradeKt` from Java